### PR TITLE
[sdk] Allow `/` in metric instrument names

### LINKED ISSUE #4877: Support "/" in instrument names
# Feature Request

The spec has changed https://github.com/open-telemetry/opentelemetry-specification/pull/3684.
The expected code change should be similar to what's been done in #4798.

### PR DESCRIPTION
**Pull request by tombiddulph**
Fixes #4877
Design discussion issue #

## Changes
Updates the regex in `MeterProviderBuilderSdk.cs` to allow metric instrument names to contain `/` characters. This is in response to the issue raised in [#4887](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4877). Now, metric names like `my_metric/environment/database` can be used. Corresponding changes are also made to the tests and CHANGELOG.md.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes

### CODE REVIEW

File: src/OpenTelemetry/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+* Allowed metric instrument names to contain `/` characters. ([#4887](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4877))
```
Comment (reyang) [inline src/OpenTelemetry/CHANGELOG.md:5]:
```suggestion
* Allowed metric instrument names to contain `/` characters. ([#4882](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4882))
```

**Review by reyang — APPROVED** [review]

**Review by vishweshbankwar — APPROVED** [review]

**Comment by vishweshbankwar** [comment]
@tombiddulph Could you please resolve the conflicts. Thanks!

File: src/OpenTelemetry/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+* Allowed metric instrument names to contain `/` characters. ([#4882](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4882))
```
Comment (CodeBlanch) [inline src/OpenTelemetry/CHANGELOG.md:5]: nit: Would you move this below the `AggregatorStore` entry? Typically we do these in chronological order.

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
Thanks @tombiddulph !

**Review by Kielek — APPROVED** [review]

**Review by Yun-Ting — APPROVED** [review]

**Pull request merged**